src/python/grpcio_tests/tests/unit/_metadata_flags_test.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests metadata flags feature by testing wait-for-ready semantics"""
+
+import time
```

Comment:
Is there a way to test this that mocks out more of the lower levels. Ideally, we shouldn't have to actually run any of c-core to test that the `wait_for_ready` flag has been plumbed all the way down.

Reply:
Testing on the public API can bring us benefits:
1. It tests our contract to user is not breaking
1. Unlikely to be changed
1. Not rely on specific implementation